plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -257,3 +261,25 @@ def _get_filter_time(self, custom_config: Dict[str, int], normal_running_cutoff:
         self.logger.info(f"The following filter time will be used: {filter_time}")
 
         return filter_time
+
+    def check_rate_limit(self, state: Dict):
+        rate_limited = state.get(self.RATE_LIMIT_DATETIME)
+        now = time()
+        if rate_limited:
+            rate_limit_string = Utils.convert_epoch_to_readable(rate_limited)
+            log_msg = f"Rate limit value stored in state: {rate_limit_string}. "
+            if rate_limited > now:
+                log_msg += "Still within rate limiting period, skipping task execution..."
+                self.logger.info(log_msg)
+                return [], state, False, 200, None
```

Comment:
The only thing about this is that the customer won't have any idea they're rate limited 🤔

Reply:
unless we put it the task into error state, we currently have no way to display any task info back to idr

Reply:
If we return a recoverable error message/status code, these should get bubbled in the 'warning' text colour and the integration is still scheduled for the next run

Reply:
mIght be a PM question to confirm as we haven't done this before, but  I agree with Nathanael it would be good to let the customer know they're being rate limited. If we return status code 429, and then the error with cause with something like 'Rate limiting enforced from Mimecast' would show on the UI but let the integration be scheduled on the 5 minute interval, and it could prevent tickets raised for why no data has been pulled?

Reply:
The only thing is that if we put the task into an error state we won't store the fact that we're rate limited 🤔

Reply:
I've updated this now. The state should include a new rate limit value.

Reply:
Based on chat with @nmccullagh-r7 . Update plugin to return 200 on first go, immediately retry and return the new 429. This way the state will be updated to include the retry limit timestamp.